### PR TITLE
fix build issues on x86 and 32-bit arm

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

PTAL @mrunalp @umohnani8 

**- What I did**

Added build constraints on `oci/stats.go` and a new `oci/stats_32.go` for 32 bit.

**- How to verify it**

build failures without the change: https://koji.fedoraproject.org/koji/taskinfo?taskID=35082648
build successful with the change: https://koji.fedoraproject.org/koji/taskinfo?taskID=35083204

**- Description for the changelog**

fix build issues on x86 and 32-bit arm

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: oci/stats_32.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// If the configured memory limit is larger than the total memory on the sys, the
+// physical system memory size is returned
+func getMemLimit(cgroupLimit uint64) uint64 {
+	si := &syscall.Sysinfo_t{}
+	err := syscall.Sysinfo(si)
+	if err != nil {
+		return cgroupLimit
+	}
+
+	physicalLimit := uint64(si.Totalram)
```
Comment (giuseppe) [inline oci/stats_32.go]: could we made this change to stats.go without adding the new file?

Comment (lsm5) [reply oci/stats_32.go]: hmm, let me try that ..

Comment (lsm5) [reply oci/stats_32.go]: done .. PTAL..

Comment (giuseppe) [reply oci/stats_32.go]: thanks!

**Comment by giuseppe** [comment]
LGTM

**Comment by haircommander** [comment]
LGTM

**Comment by rhatdan** [comment]
Lint does not like this.
golangci-lint run --build-tags="  libdm_no_deferred_remove   seccomp selinux  containers_image_ostree_stub"
oci/stats.go:57:25: unnecessary conversion (unconvert)
	physicalLimit := uint64(si.Totalram)
	                       ^
Makefile:79: recipe for target 'lint' failed
make: *** [lint] Error 1
The command "make lint" exited with 2.

**Comment by lsm5** [comment]
> Lint does not like this.
> golangci-lint run --build-tags=" libdm_no_deferred_remove seccomp selinux containers_image_ostree_stub"
> oci/stats.go:57:25: unnecessary conversion (unconvert)
> physicalLimit := uint64(si.Totalram)
> ^
> Makefile:79: recipe for target 'lint' failed
> make: *** [lint] Error 1
> The command "make lint" exited with 2.

hmm, should I revert to a separate 32-bit file?

**Comment by haircommander** [comment]
@lsm5 @giuseppe we could split the files, or just mark it no lint (add `//nolint:unconvert` to the line) and add a comment why. I agree the extra file is a little excessive for this one conversion

**Comment by lsm5** [comment]
> @lsm5 @giuseppe we could split the files, or just mark it no lint (add `//nolint:unconvert` to the line) and add a comment why. I agree the extra file is a little excessive for this one conversion

done, PTAL

**Comment by haircommander** [comment]
LGTM

**Comment by saschagrunert** [comment]
Do we want to backport this fix to the master as well?

**Comment by rhatdan** [comment]
Wouldn't that be a forwardport.  :^)  I would say yes this should be in master.

**Comment by lsm5** [comment]
On Wed, May 29, 2019 at 05:17:50AM -0700, Daniel J Walsh wrote:
> Wouldn't that be a forwardport.  :^)  I would say yes this should be in master.

ack, i'll add to master..
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/cri-o/cri-o/pull/2409#issuecomment-496911835

-- 
Lokesh
IRC, GitHub: lsm5
GPG: 0xC7C3A0DD
https://keybase.io/lsm5

**Comment by haircommander** [comment]
too late @lsm5 https://github.com/cri-o/cri-o/pull/2420 :)

**Comment by saschagrunert** [comment]
> too late @lsm5 #2420 :)

I'm sorry :innocent:

**Pull request merged**